src/maestral/cli.py

```diff
@@ -1494,3 +1494,113 @@ def log_level(level_name: str, config_name: str) -> None:
+) -> None:
+
+    from datetime import datetime
+    from .daemon import MaestralProxy
+
+    # Reason for rel_dbx_path: os.path.join does not like leading /
+    if not dropbox_path.startswith("/"):
+        dropbox_path = "/" + dropbox_path
+
+    def download_and_compare(m: MaestralProxy, old_rev: str, new_rev: str = None):
+        """
+        Download up to two revisions to a local temporary folder
+        and compare them with a 'diff'. Only text files are supported.
+        If an unknown file type was found, everything that doesn't match
+        'text/*', an error message gets printed.
+        """
+
+        diff = m.get_file_diff(dropbox_path, old_rev, new_rev)
+
+        def color(ind: int, line: str) -> str:
+            """
+            Color diff lines.
+            Inspiration for colors was taken from the
+            well known command 'git diff'.
+            """
+
+            if ind < 2:
+                line = click.style(line, bold=True)
+            elif line.startswith("+"):
+                line = click.style(line, fg="green")
+            elif line.startswith("-"):
+                line = click.style(line, fg="red")
+            # Don't highlight these in the intro
+            elif line.startswith("@@ "):
+                line = click.style(line, fg="cyan")
+            return line
+
+        # Color the lines
+        if not no_color:
+            diff = [color(i, l) for i, l in enumerate(diff)]
+
+        # Enter pager if diff is too long
+        if not no_pager:
+            if len(diff) > 30:
+                click.echo_via_pager("".join(diff))
+                return
+
+        click.echo("".join(diff))
+
+    with MaestralProxy(config_name, fallback=True) as m:
+        if len(rev) == 0:
+            entries = m.list_revisions(dropbox_path)
+            dates = []
+            for entry in entries:
+                cm = cast(str, entry["client_modified"])
+                dt = datetime.strptime(cm, "%Y-%m-%dT%H:%M:%S%z").astimezone()
+                field = cli.DateField(dt)
+                dates.append(field.format(40)[0])
+
+            base = cli.select(
+                message="New revision:",
+                options=dates,
+                hint="(↓ to see more)" if len(dates) > 6 else "",
+            )
+
+            if base == len(dates) - 1:
+                cli.warn("Oldest version selected as base")
+                return
+
+            to_compare = (
+                cli.select(
+                    message="Old revision:",
+                    options=dates[base + 1 :],
+                    hint="(↓ to see more)" if len(dates[base + 1 :]) > 6 else "",
+                )
+                + base
+                + 1
+            )
+
+            rev = [entries[to_compare]["rev"], entries[base]["rev"]]
+        elif len(rev) > 2:
+            cli.warn("You can only compare two revisions at a time")
+            return
+        download_and_compare(m, *rev)
```

Comment:
Printing "Loading..." to the console would be nice here because the download may take some time.